tools/imatrix/imatrix.cpp

```diff
@@ -602,6 +756,16 @@ void IMatrixCollector::save_imatrix(int32_t n_chunk) const {
 
             gguf_add_tensor(ctx_gguf, in_sum2);
             gguf_add_tensor(ctx_gguf, counts);
+
+            if (!stat.activations.empty() && activation_statistics()) {
+                const int32_t nact = (int32_t) stat.activations.size();
+                struct ggml_tensor * in_sum  = ggml_new_tensor_2d(ctx, GGML_TYPE_F32, nact / nmat, nmat);
+                ggml_format_name(in_sum, "%s.in_sum", name.c_str());
+                for (int32_t j = 0; j < nval; ++j) {
```

Comment:
```suggestion
                for (int32_t j = 0; j < nact; ++j) {
```

Use the corresponding size instead of adding an implicit size equivalence requirement here.